rosenpass/src/protocol.rs

```diff
@@ -276,6 +298,9 @@ pub struct InitiatorHandshake {
     pub tx_count: usize,
     pub tx_len: usize,
     pub tx_buf: MsgBuf,
+
+    // Cookie storage for retransmission, expires PEER_COOKIE_VALUE_EPOCH seconds after creation
+    pub cookie_value: CookieStore<COOKIE_VALUE_LEN>,
```

Comment:
This – like the biscuit key – should store two keys; a stale one which is still good for decrypting and a new one for encrypting and decrypting.

The best way to achieve this is probably to factor `[CookieStore<KEY_LEN>; 2]` out into a new structure; something like `CookeRotation` which handles erasure of stale keys, encryption, and decription operations.